### PR TITLE
[#194] Concept references should support names in locales other than the default locale

### PR DESCRIPTION
**Pull request by mseaton**
…the default locale

### CODE REVIEW

**Review by mogoodrich — APPROVED** [review]
LGTM, thanks @mseaton !

File: api/src/main/java/org/openmrs/module/initializer/api/utils/Utils.java [hunk 1]
```diff
@@ -160,17 +165,31 @@ public static Concept getConceptByMapping(String mapping, ConceptService service
...
-		if (instance == null) {
-			instance = getConceptByMapping(id, service);
+		Locale originalLocale = Context.getLocale();
+		try {
+			for (Locale locale : LocaleUtility.getLocalesInOrder()) {
+				Context.setLocale(locale);
+				Concept concept = service.getConceptByName(id);
+				if (concept != null) {
+					if (!originalLocale.equals(locale)) {
+						log.warn("Found '" + id + "' in locale '" + locale + "', not in '" + originalLocale + "'");
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/initializer/api/utils/Utils.java:183]:
```suggestion
						log.warn("Found '{}' in locale '{}', not in '{}'", id, locale, originalLocale);
```

**Comment by mogoodrich** [comment]
@mseaton  looks like a check has failed?

File: api/src/main/java/org/openmrs/module/initializer/api/utils/Utils.java [hunk 2]
```diff
@@ -180,7 +180,7 @@ public static Concept fetchConcept(String id, ConceptService service) {
 				Concept concept = service.getConceptByName(id);
 				if (concept != null) {
 					if (!originalLocale.equals(locale)) {
-						log.warn("Found '" + id + "' in locale '" + locale + "', not in '" + originalLocale + "'");
+						log.warn("Found '{}' in locale '{}', not in '{}'", id, locale, originalLocale);
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/initializer/api/utils/Utils.java:183]: fyi @ibacher  @mseaton this doesn't compile...

**Comment by mogoodrich** [comment]
@mseaton  check is due to a compile error with @ibacher 's change...

File: api/src/main/java/org/openmrs/module/initializer/api/utils/Utils.java (hunk 2 above)
Comment (mseaton) [reply api/src/main/java/org/openmrs/module/initializer/api/utils/Utils.java]: @ibacher let me know if this looks good to you or what more is needed for discussion / review to merge.

**Pull request merged**